### PR TITLE
Admin user

### PR DESCRIPTION
**Pull request by YesicaNavaB**
## Summary

Administrator user

## Trello Card

https://trello.com/c/XZnT0OVr

### CODE REVIEW

File: app/controllers/users.js [hunk 1]
```diff
@@ -26,3 +26,13 @@ exports.userList = (req, res, next) =>
     .userList(req.query)
     .then(results => res.status(200).send(results))
     .catch(next);
+
+exports.signUpAdministrator = (req, res, next) => {
+  req.body.password = bcrypt.hashSync(req.body.password, salt);
+  return servicesUser
+    .createUserAdmin(req.body)
+    .then(() => {
```
Comment (jersonlopez) [inline app/controllers/users.js]: remove brackets and return

**Review by jersonlopez — CHANGES_REQUESTED** [review]

File: app/middlewares/users.js [hunk 2]
```diff
@@ -54,15 +57,25 @@ exports.signInMiddleware = [
   validateRequest
 ];
 
+exports.isAdministratorMiddleware = [
+  (req, res, next) => {
+    if (req.body.decode.administrator !== true) {
+      throw error.databaseError('You do not have permissions to perform this operation');
```
Comment (jersonlopez) [inline app/middlewares/users.js]: is this an database error?

File: app/middlewares/users.js [hunk 3]
```diff
@@ -54,15 +57,25 @@ exports.signInMiddleware = [
...
+    }
+    return next();
```
Comment (jersonlopez) [inline app/middlewares/users.js]: unnecessary return

File: app/services/users.js [hunk 4]
```diff
@@ -6,6 +6,24 @@ const { User } = require('../models'),
...
+      logger.error(`Could not update user: ${name}`);
+      throw error.databaseError(err.message);
+    });
+
+exports.validateToken = ({ email }) =>
+  User.findAndCountAll({ where: { email } }).then(result => {
+    if (result.count !== 1) {
+      throw error.validateTokenError('invalid Token ');
+    }
+  });
```
Comment (jersonlopez) [inline app/services/users.js]: Missing .catch, what happens if the db or findOne fails?

File: test/users.test.js [hunk 5]
```diff
@@ -198,3 +199,59 @@ describe('user list test', () => {
...
+            .set({ Accept: 'application/json', Authorization: tokenResponse.token })
+            .then(result => {
+              expect(result.statusCode).toBe(201);
+              expect(result.text).toBe('the administrator user was created correctly');
+              dictum.chai(response, 'should register administrator with all the fields correctly');
+              done();
+            });
+        })
+    );
+  });
```
Comment (jersonlopez) [inline test/users.test.js:230]: remember \n jajja :D

File: test/utils.test.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  {
+    afterCreate: model => {
+      model.password = bcrypt.hashSync(model.password, salt);
+      return model.save();
+    }
+  }
+);
+
+exports.factoryCreate = ({ name, lastName, email, password, administrator }) =>
+  factory.create('User', { name, lastName, email, password, administrator }).then(user => user);
```
Comment (jersonlopez) [inline test/utils.test.js]: what is doing the .then?

**Review by jersonlopez — CHANGES_REQUESTED** [review]

File: test/utils.test.js (hunk 6 above)
Comment (YesicaNavaB) [reply test/utils.test.js]: Nothing :fearful:

**Review by jersonlopez — APPROVED** [review]
good job!! 💯 🥇

**Pull request merged**